business/iter8.go

```diff
@@ -0,0 +1,279 @@
+	for _, item := range result.Items {
+
+		experiments  = append (experiments,  models.ExperimentListItem{
+			Name: item.Name,
+			Phase: string(item.Status.Phase),
+			Status: item.Status.Message,
+			CreatedAt: formatTime(item.CreationTimestamp.Time),
+			Baseline: item.Spec.TargetService.Baseline,
+			BaselinePercentage: item.Status.TrafficSplit.Baseline,
+			Candidate: item.Spec.TargetService.Candidate,
+			CandidatePercentage: item.Status.TrafficSplit.Candidate,
+			Namespace: namespace,
+		})
+	}
+	return experiments, nil
+}
+
+func (in *Iter8Service) Iter8ExperimentCreate(body []byte) ( *iter8v1alpha1.Experiment, error){
+	result := &iter8v1alpha1.Experiment{}
+	k8sConfig, err := kubernetes.ConfigClient()
+	if err != nil {
+		return nil, err
+		// return models.ExperimentList{}, err
+	}
+
+	cacheToken := ""
+	kConfig := kialiConfig.Get()
+	if kConfig.InCluster {
+		if saToken, err := kubernetes.GetKialiToken(); err != nil {
+			return nil,  err
+			// return models.ExperimentList{}, err
+		} else {
+			cacheToken = saToken
+		}
+	}
+	log.Infof("Found token len %d", len(cacheToken))
+	k8sConfig.QPS = config.Get().KubernetesConfig.QPS
+	k8sConfig.Burst = config.Get().KubernetesConfig.Burst
+	k8sConfig.BearerToken = cacheToken
+	k8s, err := kube.NewForConfig(k8sConfig)
+	newexperimentSpec := &models.ExperimentSpec{}
+	err2 := json.Unmarshal(body, newexperimentSpec)
+	if err2 != nil {
+		log.Errorf("JSON: %s shows error: %s", string(body), err2)
+		err := fmt.Errorf("Bad Experiment json")
+		return nil, err
+	}
+	experiment := iter8v1alpha1.Experiment {
+		TypeMeta: metav1.TypeMeta {
+			APIVersion: iter8v1alpha1.SchemeGroupVersion.String(),
+			Kind: "Experiment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      newexperimentSpec.Name,
+			Namespace: newexperimentSpec.Namespace,
+		},
+		Spec: iter8v1alpha1.ExperimentSpec {
+			TargetService: iter8v1alpha1.TargetService {
+				ObjectReference: &corev1.ObjectReference {
+					Name: newexperimentSpec.Service,
+					Namespace: "bookinfo-iter8",
+					APIVersion: "v1",
+				},
+				Baseline: newexperimentSpec.Baseline,
+				Candidate: newexperimentSpec.Candidate,
+			},
+
+			TrafficControl: iter8v1alpha1.TrafficControl {
+				Strategy: &newexperimentSpec.TrafficControl.Algorithm,
+				MaxTrafficPercentage: &newexperimentSpec.TrafficControl.MaxTrafficPercentage,
+				TrafficStepSize : &newexperimentSpec.TrafficControl.TrafficStepSize,
+				Interval: &newexperimentSpec.TrafficControl.Interval,
+				MaxIterations: &newexperimentSpec.TrafficControl.MaxIteration,
+			},
+
+
+		},
+	}
+	log.Infof("Ready to create %s", string(body))
+
+	path := fmt.Sprintf("/apis/iter8.tools/v1alpha1/")
+	var timeout time.Duration
+	err = k8s.CoreV1().RESTClient().Post().AbsPath(path).
+		Namespace(newexperimentSpec.Namespace).
+		Resource("experiments").
+		Body(&experiment).
+		Timeout(timeout).
+		Do().
+		Into(result)
+	if err != nil {
+		log.Infof("Create return error %s", err)
+	}
+	return result, err
+}
+
```

Comment:
Okey, I think this class is interesting.
Feature wise, functionality is clear what it tries to do.
Also it's good that this class has triggered some questions in kiali backend, mostly about how to organize REST client declaration and where to put some code (it's good, I expected this).

So, @jadeyliu939 I suggest that I can work on this PR and try to address some of my comments related to re-organization.

Nothing wrong with the PR, just as this is the first extension added in Kiali from a different API and perhaps it requires some previous organization in Kiali in order to add it.

Reply:
Thank you. Sorry about the "not so clean" code :-) And thanks for taking the time to look thru it. But I am not sure what the next step, do i fix the code according to your comments? I can commit to my branch per your comments ( one commit per comment?) And do another PR with detail about what I have done?  ( Sorry, not very familiar with Git's bell and whistle).

Reply:
Hi @jadeyliu939, I propose you that I'm going to send you a PR to your branch with my suggestions, then if you are agree, we can consolidate them on this PR and merge together once is ok.